plugins/ecs-bridge/e2eTests/e2e_test.go

```diff
@@ -0,0 +1,326 @@
+// +build e2e
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package e2eTests
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"runtime"
+	"strconv"
+	"testing"
+
+	"github.com/containernetworking/cni/pkg/invoke"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	bridgeName            = "ecs-test-bridge"
+	ifName                = "ecs-test-eth0"
+	containerID           = "contain-er"
+	expectedBridgeAddress = "169.254.172.1/22"
+	expectedGateway       = "169.254.172.1"
+	expectedVethAddress   = "169.254.172.2/22"
+	dst                   = "169.254.170.2/32"
+	netConf               = `
+{
+    "type":"ecs-bridge",
+    "cniVersion":"0.3.0",
+    "bridge":"%s",
+    "ipam":{
+	"type":"ecs-ipam",
+	"id":"test",
+	"cniVersion":"0.3.0",
+	"ipv4-subnet":"169.254.172.0/22",
+	"ipv4-routes":[
+	    {
+		"dst":"%s"
+	    }
+	]
+    }
+}`
+)
+
+func init() {
+	runtime.LockOSThread()
+}
+
+// configureNetNSFunc function type defines a method that configures the network
+// namespace before executing the "ADD" command
+type configureNetNSFunc func() error
+
+func TestAddDel(t *testing.T) {
+	testCases := map[string]configureNetNSFunc{
+		"When Bridge Exists":                                configureNetNSWithBridge,
+		"When Bridge Exists And Configured With IP Address": configureNetNSWithBridgeAndSetIPAddress,
+		"When Bridge Does Not Exist":                        configureNetNSNop,
+	}
+
+	// Ensure that the bridge plugin exists
+	bridgePluginPath, err := invoke.FindInPath("ecs-bridge", []string{os.Getenv("CNI_PATH")})
+	require.NoError(t, err, "Unable to find bridge plugin in path")
+
+	// Create a directory for storing test logs
+	testLogDir, err := ioutil.TempDir("", "ecs-bridge-e2e-test-")
+	require.NoError(t, err, "Unable to create directory for storing test logs")
+
+	// Configure the env var to use the test logs directory
+	os.Setenv("ECS_CNI_LOG_FILE", fmt.Sprintf("%s/bridge.log", testLogDir))
+	t.Logf("Using %s for test logs", testLogDir)
+	defer os.Unsetenv("ECS_CNI_LOG_FILE")
+
+	// Handle deletion of test logs at the end of the test execution if
+	// specified
+	ok, err := strconv.ParseBool(getEnvOrDefault("ECS_PRESERVE_E2E_TEST_LOGS", "false"))
```

Comment:
Probably you can use this `Failed()` [method](https://golang.org/pkg/testing/#B.Failed) in the defer function. like this:
```
defer func(){
    if !t.Failed(){
       os.RemoveAll(testLogDir)
    }
}()
```